UDF/DbFunction support

nhatier opened the pull request:
* UDF/DbFunction support
 * Allows calling functions such as System.Data.Entity.SqlServer.SqlFunctions.DateDiff (issue #31), or UDF, any function declared with DbFunctionAttribute
* Ampersand can be used both as logical And or as vb-like concatenation operator
* Allow string concatenation with non-strings
* isnull sql function "isnull(a, b)"

NArnott commented:
Thanks for the contribution.  Before I test and merge this, can you update the doc xml files to reflect your changes?

nhatier commented:
I would be happy to, but 
1. I will need minimal guidance as this is not something I'm used to (which file(s)..?)
2. Considering my English is not native, the wording may not be perfect.

NArnott commented:
I'd be happy to fix any grammar or other issues with the docs after your initial attempt.  Just as the developer, it would be best if you give it a shot first so I don't totally mess up your intent.

The web doc is dynamically generated from various xml files located in `Src\DynamicLinqWebDocs\App_Data` (in the DynamicLinqWebDocs project).  I'd suggest looking at the live version at http://dynamiclinq.azurewebsites.net and figure out where you'd like to make changes, and then find the corresponding xml.

nhatier commented:
I added basic documentation of isnull in Keywords
Same thing for the Ampersand operator, as there was no Operators section

I added the [DbFunctionAttribute] support documentation to the Expressions section, though I'm not 100% sure it belongs there. Also, the example for this one is unlikely to work as-is, at it requires some backing classes, configuration and SQL functions.

StefH commented:
@nhatier  ; the isnull method can be simplified by using Coalesce:

```
Expression ParseIsNull()
{
	int errorPos = _token.pos;
	NextToken();
	Expression[] args = ParseArgumentList();
	if (args.Length != 2)
		throw ParseError(errorPos, Res.IsNullRequiresTwoArgs);

	return Expression.Coalesce(args[0], args[1]);
}
```
For details + unit-tests see https://github.com/StefH/System.Linq.Dynamic.Core/issues/13
